Sources/VLCMedia.m

```diff
@@ -769,7 +764,12 @@ - (NSString *)metadataForKey:(NSString *)key
     if (!p_md)
         return nil;
 
-    char *returnValue = libvlc_media_get_meta(p_md, [VLCMedia stringToMetaType:key]);
+    libvlc_meta_t meta = [VLCMedia stringToMetaType:key];
+    if (meta == libvlc_meta_invalid) {
```

Comment:
Right now, if the user asks for a meta data with an invalid key, unrecoverable run-time error occurs from accessing a non-existent key in the dictionary.  The addition instead checks if the user's query is valid and raises an exception instead of crashing.